sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuJsonTuple.scala

```diff
@@ -50,53 +49,52 @@ case class GpuJsonTuple(children: Seq[Expression]) extends GpuGenerator
     }
   }
 
-  def generate(inputBatch: ColumnarBatch,
+  override def generate(inputBatch: ColumnarBatch,
       generatorOffset: Int,
       outer: Boolean): ColumnarBatch = {
-
-    val jsonGpuExpr = GpuBindReferences.bindReference(jsonExpr, 
-      Seq(jsonExpr.asInstanceOf[NamedExpression].toAttribute))
+    
+    val json = inputBatch.column(generatorOffset).asInstanceOf[GpuColumnVector].getBase
     val schema = Array.fill[DataType](fieldExpressions.length)(StringType)
+    val nonGeneratorCols = (0 until generatorOffset).map { i =>
```

Comment:
nit: This map should be a safeMap as it is returning columns and if an exception is thrown the ones already produce will be leaked.

Also there should be an `closeOnExcept` after this because if an exception happens before line 78 all of these columns will also be leaked. It probably would be better to just move this down closer to where it is used instead of generating it so early.

Reply:
TIL, thank you! Changes updated.